pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -107,7 +146,7 @@ func (c *Controller) Services() ([]*model.Service, error) {
 // GetService retrieves a service by hostname if exists
 func (c *Controller) GetService(hostname model.Hostname) (*model.Service, error) {
 	var errs error
-	for _, r := range c.registries {
+	for _, r := range c.GetRegistries() {
```

Comment:
If the controller itself gets terminated due to delete, the behavior onwards will be undefined if I'm not mistaken.

Reply:
Aggregate controller will terminate only on pilot restart. For dynamically created controllers, secret controller will take care of stopping them.